src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupService.java

```diff
@@ -97,6 +97,40 @@ public void setTransientPropertiesForUser(User user, TutorialGroup tutorialGroup
             tutorialGroup.setTeachingAssistantName(null);
         }
         this.setNextSession(tutorialGroup);
+        this.setAverageAttendance(tutorialGroup);
+    }
+
+    /**
+     * Sets the averageAttendance transient field of the given tutorial group
+     * <p>
+     * Calculation:
+     * <ul>
+     * <li>Get set of the last three completed sessions (or less than three if not more available)</li>
+     * <li>Remove sessions without attendance data (null) from the set</li>
+     * <li>If set is empty, set attendance average of tutorial group to null (meaning could not be determined)</li>
+     * <li>If set is non empty, set the attendance average of the tutorial group to the arithmetic mean (rounded to integer)</li>
+     * </ul>
+     *
+     * @param tutorialGroup the tutorial group to set the averageAttendance for
+     */
+    private void setAverageAttendance(TutorialGroup tutorialGroup) {
+        Collection<TutorialGroupSession> sessions;
+        if (getPersistenceUtil().isLoaded(tutorialGroup, "tutorialGroupSessions") && tutorialGroup.getTutorialGroupSessions() != null) {
```

Comment:
Why don't you use Hibernate.isInitialized(), that way future refactoring does not have to care about these string literals

Reply:
Persistence.getPersistenceUtil is just more generic as if you can change JPA implementation underneath without changing your code.